lib/block_view_lockups.go

```diff
@@ -739,16 +1075,36 @@ func (bav *UtxoView) _connectCoinLockup(
 			errors.Wrap(RuleErrorCoinLockupInvalidLockupDuration, "_connectCoinLockup")
 	}
 
-	// Compute the lockup duration in nanoseconds.
-	lockupDurationNanoSeconds := txMeta.UnlockTimestampNanoSecs - blockTimestampNanoSecs
+	// Validate the vesting end timestamp as expiring at a logically valid time.
+	if txMeta.VestingEndTimestampNanoSecs < txMeta.UnlockTimestampNanoSecs {
```

Comment:
Isn't VestingEndTimestampNanoSecs zero in the case of a non-vesting lockup? It seems like this check would then fail for all non-vesting lockups. Not sure if I'm missing something.

Reply:
Nevermind. I figure dout they're equal in the case of a "point" lockup. I would document this in the definition of the metadata, as I mentioned in a prior comment.